File: Iniciante/1036 - Fórmula de Bhaskara/Python/1036.py
import math


def main():
    a, b, c = map(float, input().split())
    delta = calculate_delta(a, b, c)
    if delta < 0 or a == 0:
        print("Impossivel calcular")
    else:
        r1, r2 = calculate_roots(a, b, delta)
        print(f"R1 = {r1:.5f}")
        print(f"R2 = {r2:.5f}")


def calculate_delta(a, b, c):
    return b**2 - (4 * a * c)


def calculate_roots(a, b, delta):
    r1 = (-b + math.sqrt(delta)) / (2 * a)
    r2 = (-b - math.sqrt(delta)) / (2 * a)
    return r1, r2


if __name__ == "__main__":
    main()

File: Iniciante/1036 - Fórmula de Bhaskara/TypeScript/1036.ts
import * as fs from "fs";

function main(): void {
  const input: string = fs.readFileSync("/dev/stdin", "utf8");
  const [a, b, c] = input.split(" ").map(Number);
  const delta: number = calculateDelta(a, b, c);
  if (delta < 0 || a === 0) {
    console.log("Impossivel calcular");
  } else {
    const roots: number[] = calculateRoots(a, b, delta);
    console.log(`R1 = ${roots[0].toFixed(5)}`);
    console.log(`R2 = ${roots[1].toFixed(5)}`);
  }
}

function calculateDelta(a: number, b: number, c: number): number {
  return Math.pow(b, 2) - 4 * a * c;
}

function calculateRoots(a: number, b: number, delta: number): number[] {
  const r1 = (-b + Math.sqrt(delta)) / (2 * a);
  const r2 = (-b - Math.sqrt(delta)) / (2 * a);
  return [r1, r2];
}

main();
